cumulusci/robotframework/pageobjects/__init__.py

```diff
@@ -0,0 +1,17 @@
+from robot.libraries.BuiltIn import BuiltIn
+from .PageObjects import PageObjects  # noqa: F401
+from .baseobjects import BasePage, ListingPage, HomePage, DetailPage  # noqa: F401
+
+
+def pageobject(pageType, object_name=None):
+    """A decorator to designate a class as a page object"""
+    BuiltIn().log("importing page object {} {}".format(pageType, object_name), "DEBUG")
+
+    def wrapper(cls):
+        key = (pageType, object_name)
+        PageObjects.registry[key] = cls
+        if not hasattr(cls, "object_name"):
+            cls.object_name = object_name
```

Comment:
Should probably raise an exception if it has an object_name and it's different.

Reply:
Excellent observation, thanks for making the comment.

I think for now I'm going to leave it as is. My main concern is that this is in a decorator, and decorators usually shouldn't raise exceptions. I think that's debatable, though, and I could be swayed either way with nothing more than a good pint of beer :-)

For now I'm going to leave it as is since users probably won't stumble on it, but I'll make a note to mention in the documentation what happens when you do this. I don't want this issue to hold this code up.

Reply:
I think a decorator should raise an exception if you're trying to use it for something that doesn't make sense. This feels like a check we can easily add to prevent a hard-to-debug mistake. I'm less worried about it if we stop recommending to set object_name both as a class attribute and as an argument to the decorator, though.